File: frontend/src/modules/admin/types/course.ts
/**
 * 课程相关类型定义
 */

/**
 * 课程信息
 */
export interface Course {
  id: number;
  courseCode: string;  // 课程编号
  courseName: string;  // 课程名称
  departmentId: number;  // 开课院系ID
  departmentName: string;  // 开课院系名称
  teacherId: number;  // 教师ID
  teacherName: string;  // 教师名称
  credits: number;  // 学分
  courseType: string;  // 课程类型: REQUIRED(必修), ELECTIVE(选修), PUBLIC(公共课)
  semester: string;  // 学期，例如: 2023-2024-1
  capacity: number;  // 容量
  selectedCount: number;  // 已选人数
  status: string;  // 课程状态: NOT_STARTED(未开始), IN_PROGRESS(进行中), ENDED(已结束)
  description?: string;  // 课程描述
  classTime: string;  // 上课时间
  classroom: string;  // 上课地点
  weeks: string;  // 上课周次
  sections: string;  // 上课节次
}

/**
 * 课程表单数据
 */
export interface CourseForm {
  id?: number;  // 课程ID，新增时不需要
  courseCode: string;  // 课程编号
  courseName: string;  // 课程名称
  departmentId: number;  // 开课院系ID
  teacherId: number;  // 教师ID
  credits: number;  // 学分
  courseType: string;  // 课程类型
  semester: string;  // 学期
  capacity: number;  // 容量
  status: string;  // 课程状态
  description?: string;  // 课程描述
  classTime: string;  // 上课时间
  classroom: string;  // 上课地点
  weeks: string;  // 上课周次
  sections: string;  // 上课节次
}

/**
 * 院系信息
 */
export interface Department {
  id: number;  // 院系ID
  name: string;  // 院系名称
  code: string;  // 院系编码
}

/**
 * 教师信息
 */
export interface Teacher {
  id: number;  // 教师ID
  realName: string;  // 教师姓名
  department: string;  // 所属院系
}

/**
 * 课程类型枚举
 */
export enum CourseType {
  REQUIRED = 'REQUIRED',  // 必修课
  ELECTIVE = 'ELECTIVE',  // 选修课
  PUBLIC = 'PUBLIC'  // 公共课
}

/**
 * 课程状态枚举
 */
export enum CourseStatus {
  NOT_STARTED = 'NOT_STARTED',  // 未开始
  IN_PROGRESS = 'IN_PROGRESS',  // 进行中
  ENDED = 'ENDED'  // 已结束
}

/**
 * 课程类型标签映射
 */
export const courseTypeTagMap: Record<string, string> = {
  [CourseType.REQUIRED]: 'danger',
  [CourseType.ELECTIVE]: 'success',
  [CourseType.PUBLIC]: 'warning'
};

/**
 * 课程类型名称映射
 */
export const courseTypeLabelMap: Record<string, string> = {
  [CourseType.REQUIRED]: '必修课',
  [CourseType.ELECTIVE]: '选修课',
  [CourseType.PUBLIC]: '公共课'
};

/**
 * 课程状态标签映射
 */
export const courseStatusTagMap: Record<string, string> = {
  [CourseStatus.NOT_STARTED]: 'info',
  [CourseStatus.IN_PROGRESS]: 'success',
  [CourseStatus.ENDED]: 'warning'
};

/**
 * 课程状态名称映射
 */
export const courseStatusLabelMap: Record<string, string> = {
  [CourseStatus.NOT_STARTED]: '未开始',
  [CourseStatus.IN_PROGRESS]: '进行中',
  [CourseStatus.ENDED]: '已结束'
}; 

File: backend/src/main/java/com/example/campussysteam/module/course/entity/Course.java
package com.example.campussysteam.module.course.entity;

import com.example.campussysteam.module.student.entity.Department;
import com.example.campussysteam.module.user.entity.User;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 课程实体类
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "course")
public class Course {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    /**
     * 课程编号
     */
    @Column(nullable = false, unique = true, length = 20)
    private String courseCode;
    
    /**
     * 课程名称
     */
    @Column(nullable = false, length = 50)
    private String courseName;
    
    /**
     * 开课院系
     */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "department_id", nullable = false)
    private Department department;
    
    /**
     * 教师ID
     */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "teacher_id", nullable = false)
    private User teacher;
    
    /**
     * 教师姓名（冗余字段，便于展示）
     */
    @Column(length = 50)
    private String teacherName;
    
    /**
     * 学分
     */
    @Column(nullable = false)
    private Double credits;
    
    /**
     * 课程类型: REQUIRED(必修), ELECTIVE(选修), PUBLIC(公共课)
     */
    @Column(nullable = false, length = 20)
    private String courseType;
    
    /**
     * 学期，例如: 2023-2024-1
     */
    @Column(nullable = false, length = 20)
    private String semester;
    
    /**
     * 上课时间
     */
    @Column(length = 100)
    private String classTime;
    
    /**
     * 上课地点
     */
    @Column(length = 100)
    private String classroom;
    
    /**
     * 上课周次
     */
    @Column(length = 50)
    private String weeks;
    
    /**
     * 上课节次
     */
    @Column(length = 50)
    private String sections;
    
    /**
     * 课程容量
     */
    @Column(nullable = false)
    private Integer capacity;
    
    /**
     * 已选人数
     */
    @Column(nullable = false)
    @Builder.Default
    private Integer selectedCount = 0;
    
    /**
     * 课程状态: NOT_STARTED(未开始), IN_PROGRESS(进行中), ENDED(已结束)
     */
    @Column(nullable = false, length = 20)
    private String status;
    
    /**
     * 课程描述
     */
    @Column(length = 500)
    private String description;
    
    /**
     * 创建时间
     */
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    /**
     * 更新时间
     */
    @Column(nullable = false)
    private LocalDateTime updatedAt;
    
    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
} 